filebeat/input/tcp/input.go

```diff
@@ -110,9 +99,12 @@ func (s *server) Run(ctx input.Context, publisher stateless.Publisher) error {
 	log.Info("starting tcp socket input")
 	defer log.Info("tcp input stopped")
 
+	reg, unreg := inputmon.NewInputRegistry("tcp", ctx.ID, nil)
+	defer unreg()
+
```

Comment:
This seems like an ergonomics loss to me. Same for UDP.

Reply:
ok if I interpret the "ergonomics loss" you refer to the `defer unreg`?! If yes how would you propose to maintain the same ergonomics?! If no, where are you referring? 😄

Reply:
Add a method on TCP and UDP to get the `monitoring.Registry` so that the metrics can be added would be my suggestion.